### ISSUE TITLE
Is emulating `position:device-fixed` really the best use case for this API?

### ISSUE THREAD

**Issue by theres-waldo** [bokand/ViewportAPI#60]
We are about to ship this API in Firefox, and I'm working on coordinating some MDN documentation.

I notice that in this repo, the [headline example](https://wicg.github.io/visual-viewport/examples/fixed-to-viewport.html) is emulating `position: device-fixed`. I wonder, though -- is this the best use case for this API?

In browsers with asynchronous scrolling, which includes both Chrome and Firefox on Android, updating a position property from script in response to scrolling is going to cause flickering during scrolling. (And indeed, if I open the headline example on Android with either Chrome or Firefox, the red box flickers as you scroll.)

It feels strange to me to be encouraging web developers to author websites with flickering elements. I would feel better about promoting examples that don't involve such artifacts.

(On a related note: if we want users to be able to have `position: device-fixed` behaviour, perhaos a better approach would be to actually implement `position: device-fixed`, which then doesn't need to cause such a flickering artifact?)

**Comment by bokand** [comment]
I agree it's not a great example, it's simply the most obvious/visible one. This would definitely be better served with position: device-fixed but encouraging this use-case in general is somewhat questionable. It has accessibility and usability concerns.

The main motivation for us in visualViewport was that we wanted to make window APIs like scrollX|Y and innerWidth|Height relative to the layout viewport. We had a long tail of bugs when pinch-zooming as a result of having these be relative to the visual viewport. When we first made this change, some authors complained about losing the ability to tell the viewport was zoomed (https://crbug.com/571297). The most compelling cases where things like "Remote Viewer App" (seeing when the remote user is zoomed in), and analytics ("how often do users zoom on my site"). These are fairly niche but on the principle of a rational platform we agreed that these things should be possible.
